src/operator/convolution.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+import edgeHandlingMethod from './edgeHandlingMethod'
+
+//convolution using a square kernel
+export default function convolution(newImage, kernel, edgeHandling){
+    let kernelWidth = Math.sqrt(kernel.length);
+    let k = Math.floor(kernelWidth/2);
+    let div = 0;
+    edgeHandling = edgeHandling || 'copy';
+
+    for(let i = 0; i < kernel.length; i++){div += kernel[i];}
+
+    for(let x = 0; x < this.width ; x++){
+        for(let y = 0; y < this.height ; y++){
+            let sum = 0;
+            for(let i = -k; i <= k; i++){
+                for(let j = -k; j <= k; j++){
+                    let val = edgeHandling.toLowerCase() == 'mirror'
+                        ? mirrorValue(x, y, i, j, this)
+                        : (isOutSidePixel(x, y, this) ? undefined : this.getValueXY(x + i, y + j, 0));
+                    if(val != undefined)
+                        sum += val*kernel[(i + k)*kernelWidth + (j + k)];
+                }
+            }
+            let newValue;
+            if(div >= 1){
+                newValue = Math.floor(sum/div);
+            }else{
+                newValue = sum;
+            }
+
+            newImage.setValueXY(x, y, 0, newValue);
+            if(this.alpha){
+                newImage.setValueXY(x, y, 1, this.getValueXY(x, y, 1));
+            }
+        }
+    }
+
+    if(edgeHandling.toLowerCase() != 'mirror'){
```

Comment:
please use the triple equal operators (=== and !==)
In javascript you are able to compare different types and there will be some implicit conversions.
For example (true == 1) is evaluated to true. The same for ('' == false) is evaluated to true, same for ('1' == 1). This may be on one-hand misleading but also takes time for conversion.
This is why the triple equals is useful. Before making any comparison, the type must be the same.